Chef provisioner fails without root

```
<snip>
    virtualbox-iso: Preparing...                ########################################### [100%]
    virtualbox-iso: 1:chef                   ########################################### [100%]
    virtualbox-iso: Thank you for installing Chef!
    virtualbox-iso: Creating directory: /tmp/packer-chef-client
    virtualbox-iso: Uploading validation key...
==> virtualbox-iso: Unregistering and deleting virtual machine...
==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored: Error copying validation key: scp: /tmp/packer-chef-client/validation.pem: Permission denied

==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Error copying validation key: scp: /tmp/packer-chef-client/validation.pem: Permission denied

==> Builds finished but no artifacts were created.
make: *** [vm] Error 1
```

`ChefClient.createDir` uses sudo to create the `StagingDir`, creating it with owner/group `root:root`, but then `packer.Communicator` uses the user defined in the manifest to upload the validator key, which fails if that user is not `root`.

I think the `prevent_sudo` option should probably be applied in `createDir` to deal with this. Would this be an acceptable solution?

--- a/provisioner/chef-client/provisioner.go
+++ b/provisioner/chef-client/provisioner.go
@@ -310,16 +310,25 @@ func (p *Provisioner) createDir(ui packer.Ui, comm packer.Communicator, dir stri
 		mkdirCmd = "sudo " + mkdirCmd
 	}
 
-	cmd := &packer.RemoteCmd{
-		Command: mkdirCmd,
-	}
-
+	cmd := &packer.RemoteCmd{Command: mkdirCmd}
 	if err := cmd.StartWithUi(comm, ui); err != nil {
 		return err
 	}
-
 	if cmd.ExitStatus != 0 {
-		return fmt.Errorf("Non-zero exit status.")
+		return fmt.Errorf("Non-zero exit status. See output above for more info.")
+	}
+
+	// Chmod the directory to 0777 just so that we can access it as our user
+	mkdirCmd = fmt.Sprintf("chmod 0777 '%s'", dir)
+	if !p.config.PreventSudo {
+		mkdirCmd = "sudo " + mkdirCmd
+	}
+	cmd = &packer.RemoteCmd{Command: mkdirCmd}
+	if err := cmd.StartWithUi(comm, ui); err != nil {
+		return err
+	}
+	if cmd.ExitStatus != 0 {
+		return fmt.Errorf("Non-zero exit status. See output above for more info.")
 	}
 
 	return nil

--- a/website/source/docs/provisioners/chef-client.html.markdown
+++ b/website/source/docs/provisioners/chef-client.html.markdown
@@ -161,3 +161,12 @@ curl -L https://www.opscode.com/chef/install.sh | \
 ```
 
 This command can be customized using the `install_command` configuration.
+
+## Folder Permissions
+
+The `chef-client` provisioner will chmod the directory with your Chef
+keys to 777. This is to ensure that Packer can upload and make use of that
+directory. However, once the machine is created, you usually don't
+want to keep these directories with those permissions. To change the
+permissions on the directories, append a shell provisioner after Chef
+to modify them.

--- a/website/source/docs/provisioners/chef-client.html.markdown
+++ b/website/source/docs/provisioners/chef-client.html.markdown
@@ -164,7 +164,7 @@ This command can be customized using the `install_command` configuration.
 
 ## Folder Permissions
 
-The `chef-client` provisioner will chmod the directory with your Chef
+!> The `chef-client` provisioner will chmod the directory with your Chef
 keys to 777. This is to ensure that Packer can upload and make use of that
 directory. However, once the machine is created, you usually don't
 want to keep these directories with those permissions. To change the

